tests/behat/features/api/get_scratch_remix_programs.feature

```diff
@@ -0,0 +1,125 @@
+  Catrobat language version 0.999 should correctly add remix relations
+    #-------------------------------------------------------------------------------------------------------------------
+    # this is how the remix graph should look like after the program is uploaded:
+    #
+    #       (SCRATCH #1)  (SCRATCH #2)
+    #                \      /
+    #                \     /
+    #                 (10)            <-- to be added (uploaded program will get ID "10")
+    #
+    #-------------------------------------------------------------------------------------------------------------------
+    Given I have a program with "catrobatLanguageVersion" set to "0.999" and "url" set to "Music Inventor [https://scratch.mit.edu/projects/29495624], The Colour Divide - Trailer [https://scratch.mit.edu/projects/70058680/]"
+    When I upload the program with the id "10"
+    Then the uploaded program should be a remix root
+    And the uploaded program should have remix migration date NOT NULL
+    And the uploaded program should have a Catrobat forward ancestor having its own id and depth "0"
+    And the uploaded program should have no Catrobat ancestors except self-relation
+    And the uploaded program should have a Scratch parent having id "29495624"
+    And the uploaded program should have a Scratch parent having id "70058680"
+    And the uploaded program should have no further Scratch parents
+    And the uploaded program should have no Catrobat forward descendants except self-relation
+    And the uploaded program should have RemixOf "Music Inventor [https://scratch.mit.edu/projects/29495624], The Colour Divide - Trailer [https://scratch.mit.edu/projects/70058680/]" in the xml
+    Given I have a parameter "limit" with value "6"
+    And I have a parameter "offset" with value "0"
+    When I GET "/app/api/projects/scratchRemixes.json" with these parameters
+    Then I should get 6 programs in the following order:
+      | Name      |
+      | program 5 |
+      | program 6 |
+      | program 7 |
+      | program 8 |
+      | program 9 |
+      | test      |
+  Scenario: program upload with parent-URL referring to existing Catrobat programs and
+  Catrobat language version 1.0 should correctly add remix relations (example #2)
+    Given there are Scratch remix relations:
+      | scratch_parent_id | catrobat_child_id |
+      | 29495624          | 12                 |
+      | 70058680          | 13                 |
+
+    And there are backward remix relations:
+      | parent_id | child_id |
+      | 9         | 8        |
+      | 6         | 4        |
+      | 6         | 1        |
+
+    Given I have a program with "catrobatLanguageVersion" set to "1.0" and "url" set to "The Colour Divide - Trailer [https://scratch.mit.edu/projects/70058680/], Merge 2 [Program 2 [/pocketalice/project/2], Merge 1 [Program 6 [/app/project/6], Program 8 [https://share.catrob.at/app/project/8]]]"
```

Comment:
would look cleaner if you also set the project name

Reply:
Fixed